### ISSUE TITLE
Add 'Limit of blank` calculation in LOD report

### ISSUE THREAD

**Issue by leandroradusky** [instedd/cdx#1904]
In the tables exported to NIH, we need to inform the Limit of Blank computation for the LOD report.

Its computed as follows:
- The LOD is equal to the linear regression intercept in the measured signal plot.
- The linear regression commits errors at each point (unless the machine is perfect), these errors have a distribution.
- There will be some blank samples, with zero concentration, that will have measurement results. All the results will have a distribution as well.

Finally: the intersection of these two distributions gives us the limit of blank:

![image](https://user-images.githubusercontent.com/13782680/217763245-968fd0cb-86fb-4336-875c-05664eadde33.png)

Basically, this tells us where I start to confidently distinguish between blanks and positives with my machine. 

Finally this numbers will be shown in the summary section.

**Comment by bolom** [comment]
![Capture d’écran 2023-05-25 à 16 27 24](https://github.com/instedd/cdx/assets/72155/8f796e2b-d101-4d2d-a525-3014acdaf5cf)
When I'm trying to access to report section. I've got this error


> In the tables exported to NIH, we need to inform the Limit of Blank computation for the LOD report.

in which model I should add this new attribute in Boxe?

**Comment by bolom** [comment]
@leandroradusky What do you think?

I'm using this gem

https://github.com/yoshoku/rumale

```
require 'rumale'

def calculate_lod(concentrations, signals)
  concentrations_matrix = Numo::DFloat[*concentrations.map { |c| [c] }]
  lr = Rumale::LinearModel::LinearRegression.new(fit_intercept: true)
  lr.fit(concentrations_matrix, signals)

  # The y-intercept is the Limit of Detection
  lod = lr.intercept

  # Calculate the prediction for each point
  predictions = lr.predict(concentrations_matrix)

  # Calculate the residuals (differences between the observed and predicted values)
  residuals = signals - predictions

  # Calculate the root mean square error (RMSE)
  rmse = Math.sqrt((residuals**2).sum / residuals.size)

  # If there are any blank samples, calculate the LOD and LOB based on their standard deviation
  blank_samples = signals[concentrations == 0]
  if blank_samples.size > 0
    blank_samples_mean = blank_samples.mean
    blank_samples_sd = blank_samples.stddev
    lod = [lod, 3 * blank_samples_sd].max
    lob = blank_samples_mean + 1.645 * blank_samples_sd
  else
    lob = nil
  end

  # Define the gray zone as the range between LOB and LOD
  gray_zone = lob.nil? ? nil : (lob..lod)

  return lod, rmse, lob, gray_zone
end


```

Exemple of use

```
concentrations = Samples.all.map(&:concentration)
signals =  Sample.all.map(&:measured_signal)
lod, rmse, lob, gray_zone = calculate_lod(concentrations, signals)
puts "Limit of Detection: #{lod}"
puts "Root Mean Square Error: #{rmse}"
puts "Limit of Blank: #{lob}"
puts "Gray Zone: #{gray_zone}"
```

**Comment by leandroradusky** [comment]
Hi @bolom, the `Reports` of the main menu are not where we aim to work. What we want is to work into `Samples>Reports`.

![image](https://github.com/instedd/cdx/assets/13782680/9337d1e8-ba66-4da4-85c0-d56504896a59)

The model to work on then is `SamplesReport`, where LOD & LOB, which you are calculating with this function should be added as fields in the table when the report is created. 

I've checked the function that you coded to compute this, and it is ok :+1:

**Comment by sardar-usman** [comment]
@bolom It is working fine.
